### DISCUSSION TITLE
Using isGameOver() with an already generated list of legal moves

### DISCUSSION THREAD

**Discussion by LowLevel73** [Disservin/chess-library#30] [Ideas]
I have implemented a typical negamax() algorithm and after calling my method that generates the list of legal moves, I want to check if the game is over.

The isGameOver() method also calls movegen::legalmoves().

For performance reasons, do you think it would be a good idea to implement a way to check for gameover by passing an already generated list of legal moves?

**Comment by Disservin** [comment]
Hi,

yeah that’s true but in those cases of negamax it’s always more performant if you do the check yourself. 
Meaning in negamax you anyway have to generate all moves then you loop over them, at this point you already have the information whether there are no moves left and if so if the position is in check.
